meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -1,17 +1,144 @@
 # common boot handlers
 
+# uEnv handling
 setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
 setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_bootenv 'setenv bootfirmware_version; ext4load ${devtype} ${devnum}:2 ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_getroot 'setexpr ostree_root gsub "^.*ostree=([^ ]*).*$" "\\\\1" "${bootargs}"'
+
+# Env saving
+setenv saveenv_mmc 'if test -z "${board_is_closed}"; then saveenv; fi;'
+
+# Boot firmware update helpers
+setenv load_image 'ext4load ${devtype} ${devnum}:2 ${loadaddr} ${image_path}'
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+# Import uEnv.txt
+run bootcmd_otenv
+# Get deployment sysroot absolute path
+run bootcmd_getroot
+# Check state of SECONDARY_BOOT bit
+imx_secondary_boot
+# Check if board is on closed state
+imx_is_closed
```

Comment:
This is imx specific while the file is generic to everyone using scr-fit. Either we abstract or we define a boot-common.cmd.in file that is under an "imx" folder, which will be used by default on every imx-based device due MACHINEOVERRIDES.

Reply:
I think we can address that when we start working on boot firmware support on non-imx boards

Reply:
Sure, it's easy to have it under a generic imx folder when needed.